### PR TITLE
Identity manager/check auth

### PR DESCRIPTION
**Pull request by TadevosStepanyan**

### CODE REVIEW

File: src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs [hunk 1]
```diff
@@ -11,29 +15,29 @@ public static IServiceCollection IdentityApplicationBuilderExtensions(this IServ
             services
                 .AddAuthorization(options =>
                 {
-                    options.AddPolicy("admin", policy => policy.RequireClaim("Admin"));
+                    options.AddPolicy(UserPolicy.User, policy => policy.RequireAssertion(context => context.User.IsInRole("User")));
```
Comment (Soarc) [inline src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:18]: Are this policies required? I'm pretty sure, that Authorize attribute by default uses roles

File: src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs [hunk 2]
```diff
@@ -11,29 +15,29 @@ public static IServiceCollection IdentityApplicationBuilderExtensions(this IServ
...
+                    options.AddPolicy(UserPolicy.Admin, policy => policy.RequireAssertion(context => context.User.IsInRole("Admin")));
                 })
                 .AddAuthentication(options => {
                     options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
                     options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
                 })
                 .AddJwtBearer(options =>
                 {
-                    options.Authority = "https://localhost:5123";
+                    options.Authority = "https://localhost:5001";
```
Comment (Soarc) [inline src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:27]: `Authority` and `Audience `should be passed through configuration. (via appsettings.json). In production we will override values with environmental variables.

File: src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs [hunk 3]
```diff
@@ -11,29 +15,29 @@ public static IServiceCollection IdentityApplicationBuilderExtensions(this IServ
...
                     options.TokenValidationParameters = new TokenValidationParameters
                     {
                         //ValidateIssuerSigningKey = true,
-                        //ValidateAudience = true,
-                        //ValidateIssuer = true,
-                        RoleClaimType = "role",
-                        NameClaimType = "sub",
+                        ValidateAudience = false,
+                        ValidateIssuer = true,
+                        RoleClaimType = ClaimTypes.Role,
```
Comment (Soarc) [inline src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:34]: Aspnet core default ClaimType.Role value (`http://schemas.microsoft.com/ws/2008/06/identity/claims/role`). is very different, then we are passing in JWT token. This is why Authorize attribute wasn't working for you. Use  
```
RoleClaimType = "role",
NameClaimType = "sub",
```

File: src/Shared/IdentityClient/Common/User.cs [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
         public int? UserId { get; set; }
         public int? EmployeeId { get; set; }
-        public int? AgancyId { get; set; }
+        public int? AgencyId { get; set; }
+        [DisplayName(ClaimTypes.Name)]
         public string Name { get; set; }
-        public string Email { get; set; }
+        [DisplayName(ClaimTypes.Email)]
+        public string Email { get; set; }
+        [DisplayName(ClaimTypes.Role)]
```
Comment (Soarc) [inline src/Shared/IdentityClient/Common/User.cs:15]: Remove display name attributes. We don't need it.

File: src/Shared/IdentityClient/Extensions/UserExtensions.cs [hunk 5]
```diff
@@ -1,20 +1,41 @@
...
+using Travely.IdentityClient.Authorization.Data;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public static class UserExtensions
+    {
+        public static UserInfo GetTravelyUserInfo(this HttpContext httpContext)
+        {
+            UserInfo userInfo = new UserInfo();
+            foreach(var propInfo in userInfo.GetType().GetProperties())
```
Comment (Soarc) [inline src/Shared/IdentityClient/Extensions/UserExtensions.cs:14]: This is very bad code. Why are you using reflection? Reflection usage is not justified here.

Do simple assignment. Create following 
```
public static class TravelyClaims {
   public const string Role = "role";
   public const string UserId = "sub";
   public const string AgencyId= "agency_id";
// e.t.c
// for exact claim values sync with @erick20 
}
```
and create UserInfo object in following simple manner
```
var userInfo = new UserInfo {
   UserId = Claims.FirstOrDefault(claim => claim.Type == TravelyClaims.UserId)?.Value,
   AgencyId= Claims.FirstOrDefault(claim => claim.Type == TravelyClaims.AgencyId)?.Value,
// e.t.c
}

```

**Review by Soarc — CHANGES_REQUESTED** [review]

File: src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs (hunk 3 above)
Comment (TadevosStepanyan) [reply src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:34]: at the start, I was using the way that you described but it was not working that's why I have swapped to ClaimType.Role below is the image of the HttpContext.User.Claims which is coming with ClaimType.Role
image

Comment (TadevosStepanyan) [reply src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:34]: with `RoleClaimType = ClaimTypes.Role` the `[Authorize(Roles = "User")]` works properly

Comment (Soarc) [reply src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs:34]: Ok, I will double check. For now this is fine.

File: src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs (hunk 1 above)
Comment (TadevosStepanyan) [reply src/Shared/IdentityClient/Config/ConfigServicesIdentityService.cs]: no, they are not required I have removed this

File: src/IdentityManager/WebApi/appsettings.Development.json [hunk 6]
```diff
@@ -7,7 +7,8 @@
     }
   },
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=(LocalDb)\\MSSQLLocalDB;Initial Catalog=TravelyIdentityService;Integrated Security=SSPI;AttachDBFilename=|DataDirectory|\\TravelyIdentityService.mdf"
+    "DefaultConnection": "Data Source=PCH-TADEVOSS\\MSSQLSERVER2012;Initial Catalog=TravelyIdentityService;Integrated Security=SSPI;"
```
Comment (Soarc) [inline src/IdentityManager/WebApi/appsettings.Development.json:10]: Revert this file.
Create `appsettings.local.json` and use it for local connection strings. Don't change this one.

File: src/IdentityManager/WebApi/appsettings.Production.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (Soarc) [inline src/IdentityManager/WebApi/appsettings.Production.json:1]: Revert

**Review by Soarc — CHANGES_REQUESTED** [review]

**Pull request merged**